test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -303,3 +307,132 @@ func TestKsctlGeneratedTiers(t *testing.T) {
+
+		// when
+
+		// Now let's create a Space
+		user := NewSignupRequest(awaitilities).
+			Username("featured-user").
+			Email("featured@domain.com").
+			ManuallyApprove().
+			TargetCluster(awaitilities.Member1()).
+			EnsureMUR().
+			SpaceTier("base1ns").
+			TargetCluster(memberAwait).
+			RequireConditions(wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...).
+			Execute(t)
+		// and promote that space to the ftier tier
+		tiers.MoveSpaceToTier(t, hostAwait, "featured-user", tier.Name)
+
+		// then
+
+		// Verify that the space has the feature annotation
+		space, err := hostAwait.WaitForSpace(t, user.Space.Name)
+		require.NoError(t, err)
+		require.NotEmpty(t, space.Annotations)
+		assert.Equal(t, "test-feature", space.Annotations[toolchainv1alpha1.FeatureToggleNameAnnotationKey])
+		// and CRB for the that feature has been created
+		crb := &v1.ClusterRoleBinding{}
+		crbName := fmt.Sprintf("%s-%s", user.Space.Name, "test-feature")
+		err = memberAwait.WaitForObject(t, "", crbName, crb, &v1.ClusterRoleBindingList{})
+		require.NoError(t, err)
+
+		t.Run("disable feature", func(t *testing.T) {
+			// when
+
+			// Now let's disable the feature for the Space by removing the feature annotation
+			an := space.Annotations
+			delete(an, toolchainv1alpha1.FeatureToggleNameAnnotationKey)
+			space, err := hostAwait.UpdateSpace(t, user.Space.Name, func(s *toolchainv1alpha1.Space) {
+				s.Annotations = an
+			})
+			require.NoError(t, err)
+
+			// then
+			err = memberAwait.WaitForObjectDeleted(t, "", crbName, crb)
+			require.NoError(t, err)
+
+			t.Run("re-enable feature", func(t *testing.T) {
+				// when
+
+				// Now let's re-enable the feature for the Space by restoring the feature annotation
+				an := space.Annotations
+				if an == nil {
+					an = make(map[string]string)
+				}
+				an[toolchainv1alpha1.FeatureToggleNameAnnotationKey] = "test-feature"
+				_, err := hostAwait.UpdateSpace(t, user.Space.Name, func(s *toolchainv1alpha1.Space) {
+					s.Annotations = an
+				})
+				require.NoError(t, err)
+
+				// then
+				// Verify that the CRB is back
+				crb := &v1.ClusterRoleBinding{}
+				err = memberAwait.WaitForObject(t, "", crbName, crb, &v1.ClusterRoleBindingList{})
+				require.NoError(t, err)
+			})
+		})
+	})
+}
+
+func withClusterRoleBinding(t *testing.T, otherTier *toolchainv1alpha1.NSTemplateTier, feature string) tiers.CustomNSTemplateTierModifier {
+	var tpl bytes.Buffer
+	err := template.Must(template.New("crb").Parse(viewCRB)).Execute(&tpl, map[string]interface{}{
+		"featureName": feature,
+	})
+	require.NoError(t, err)
+
+	modifiers := []tiers.TierTemplateModifier{
+		func(awaitility *wait.HostAwaitility, template *toolchainv1alpha1.TierTemplate) error {
+			clusterRB := runtime.RawExtension{
+				Raw: tpl.Bytes(),
+			}
+			template.Spec.Template.Objects = append(template.Spec.Template.Objects, clusterRB)
+			return nil
+		},
+	}
+	return tiers.WithClusterResources(t, otherTier, modifiers...)
```

Comment:
no need to define it as an array
```suggestion
	modifier := func(awaitility *wait.HostAwaitility, template *toolchainv1alpha1.TierTemplate) error {
		clusterRB := runtime.RawExtension{
			Raw: tpl.Bytes(),
		}
		template.Spec.Template.Objects = append(template.Spec.Template.Objects, clusterRB)
		return nil
	}
	return tiers.WithClusterResources(t, otherTier, modifier)
```

Reply:
Right. See https://github.com/codeready-toolchain/toolchain-e2e/pull/1008/commits/bb43234bee995e327a3789b83b8512d9822a2a03